pkg/flag/remote_flags.go

```diff
@@ -87,14 +87,14 @@ func (f *RemoteFlags) ToOptions() RemoteOptions {
 	token := viper.GetString(ServerTokenFlag)
 	tokenHeader := viper.GetString(ServerTokenHeaderFlag)
 
-	if serverAddr == "" {
+	if serverAddr == "" && listen == "" {
 		switch {
 		case len(lo.FromPtr(f.CustomHeaders)) > 0:
```

Comment:
I found we needed to fix it as well.

```suggestion
		case len(customHeaders) > 0:
```

Reply:
Already fixed in https://github.com/aquasecurity/trivy/pull/2470